airflow/providers/microsoft/azure/hooks/asb.py

```diff
@@ -0,0 +1,205 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Any, Dict, List, Optional, Union
+
+from azure.servicebus import ServiceBusClient, ServiceBusMessage, ServiceBusSender
+from azure.servicebus.management import QueueProperties, ServiceBusAdministrationClient
+
+from airflow.hooks.base import BaseHook
+
+
+class BaseAzureServiceBusHook(BaseHook):
+    """
+    BaseAzureServiceBusHook class to create session and create connection. Client ID and
+    Secrete ID are optional.
```

Comment:
>Client ID and Secrete ID are optional.

How can a user connect to Service Bus with client ID/secret? I only see a connection string auth method.

Reply:
Currently to start with I am using connection string to connect to azure service bus, later we can implement connect via client ID, secret, tenant, SAS token, etc. So instead of creating that while implementing I have added those points and add that in `get_connection_form_widgets` function, Now I removed the code forgot to remove this comment